### PR TITLE
Fix alpha byte for 32bpp pix

### PR DESCRIPTION
**Pull request by alexcohn**
fixing https://github.com/DanBloomberg/leptonica/issues/442

### CODE REVIEW

File: src/pngio.c [hunk 1]
```diff
@@ -330,7 +330,10 @@ PIXCMAP     *cmap;
                 SET_DATA_BYTE(ppixel, COLOR_RED, rowptr[k++]);
                 SET_DATA_BYTE(ppixel, COLOR_GREEN, rowptr[k++]);
                 SET_DATA_BYTE(ppixel, COLOR_BLUE, rowptr[k++]);
-                SET_DATA_BYTE(ppixel, L_ALPHA_CHANNEL, (spp == 4) ? rowptr[k++] : 255);
+                if (spp == 4)
```
Comment (rmtheis) [inline src/pngio.c]: Consider `spp == 3`. Either the condition is wrong or the comment on line 335 is wrong

**Comment by DanBloomberg** [comment]
We understand that the problem is with the java test, not the leptonica readers.

So fix the tesseract tests, and in a week when I get back from a short vacation I'll modify all the leptonica RGB readers to store 255 in the 4th byte, and also comment that no program should use the 4th byte when spp == 3

**Comment by alexcohn** [comment]
Note that I have fixed the Java tests, but this fix may be helpful also in other situations.

**Comment by Robyer** [comment]
To make this better you may update also code for BMP and possibly other formats to make this behavior consistent over whole library.

For BMP format add this on line 313 in file `bmpio.c`:
`*((l_uint8 *)pword + L_ALPHA_CHANNEL) = 255;`

**Comment by DanBloomberg** [comment]
yes, will look at all read functions on RGB images

**Pull request closed without merging**